Change Form to Query for transaction list

BreD1810 opened the pull request:
The transactions endpoint expects an `account_id` query parameter, which is not currently supplied.
This PR replaces the current form implementation with query parameters.

Attempts to address #13.

danieleades commented:
changes look good to me. I've just approved the workflow. Once that's passing it should be good to merge

danieleades commented:
note that the CI job is using *nightly* rustfmt. As in, `cargo +nightly fmt`.

BreD1810 commented:
I have also made some changes to ensure deserialisation of the Transaction response works (see [here](https://github.com/BreD1810/monzo-lib/commit/d9ec0e4f2c796a1d7ec3affb7b926a37a535329b)).
Would you like that change as part of this PR or a separate one?

danieleades commented:
> I have also made some changes to ensure deserialisation of the Transaction response works (see [here](https://github.com/BreD1810/monzo-lib/commit/d9ec0e4f2c796a1d7ec3affb7b926a37a535329b)).
> Would you like that change as part of this PR or a separate one?

A separate pull request would be perfect.

For reference, I started a discussion here- #12

danieleades commented:
i'm happy with these changes. as you point out, the 'list transactions' endpoint is still broken due the categories enum not covering all cases.
If you've got another pull request that fixes that, i'd be happy to review. I think the best solution might be to simply change the enum to a string (that's a breaking change, so that would also move this package to `0.4.0`).

PS i've fixed the code coverage job. that should work now if you rebase on master

danieleades commented on examples/list_transactions.rs:
```diff
@@ -0,0 +1,29 @@
+    let transactions = client
+        .transactions(account_id)
+        .since(Utc::now() - Duration::days(89))
+        .limit(2)
```
parameters like `limit` could be exposed as args. that's a small thing though.

BreD1810 replied:
I've added flags for both the limit and the duration.

BreD1810 commented:
> PS i've fixed the code coverage job. that should work now if you rebase on master

I have rebased. Not sure why `codecov/patch` isn't passing (I cannot see any more information when clicking "details").

> the 'list transactions' endpoint is still broken due the categories enum not covering all cases.

I'm happy to have a look at this over the next few days. I quite like your idea of adding a `Other(String)` variant.

danieleades commented:
> > PS i've fixed the code coverage job. that should work now if you rebase on master
> 
> I have rebased. Not sure why `codecov/patch` isn't passing (I cannot see any more information when clicking "details").
it's checking that the coverage of the *changed* code is at least as high as the total coverage (which is one way of checking that the total coverage is higher after this diff is merged). Since the changed code has 0% code coverage this fails. That won't stop me merging, but it's informational.
When you submit a pull request to fix the transactions deserialisation, that should include tests which allow this check to pass.
> 
> > the 'list transactions' endpoint is still broken due the categories enum not covering all cases.
> 
> I'm happy to have a look at this over the next few days. I quite like your idea of adding a `Other(String)` variant.

it's not trivial, `serde` doesn't support this out of the box you'd have to implement this yourself. You could potentially use the `#[serde(from = "FromType")]` container attribute, with the `FromType` being `String`. then you'd also implement `From<String>` for `Category`. (I'm guessing here)
